07-http_gordle/final/internal/gordle/game_internal_test.go

```diff
@@ -0,0 +1,128 @@
+package gordle
+
+import (
+	"errors"
+	"testing"
+
+	"golang.org/x/exp/slices"
+)
+
+func TestGameValidateGuess(t *testing.T) {
+	tt := map[string]struct {
+		word     string
+		expected error
+	}{
+		"nominal": {
+			word:     "GUESS",
+			expected: nil,
+		},
+		"too long": {
+			word:     "POCKET",
+			expected: ErrInvalidGuess,
+		},
+		"empty": {
+			word:     "",
+			expected: ErrInvalidGuess,
+		},
+	}
+
+	for name, tc := range tt {
+		t.Run(name, func(t *testing.T) {
+			g, _ := New([]string{"SLICE"})
+
+			err := g.validateGuess(tc.word)
+			if !errors.Is(err, tc.expected) {
+				t.Errorf("%s, expected %v, got %v", tc.word, tc.expected, err)
+			}
+		})
+	}
+}
+
+func TestSplitToUppercaseCharacters(t *testing.T) {
+	tt := map[string]struct {
+		input string
+		want  []rune
+	}{
+		"lowercase": {
+			input: "pocket",
+			want:  []rune("POCKET"),
+		},
+	}
+	for name, tc := range tt {
+		t.Run(name, func(t *testing.T) {
+			got := splitToUppercaseCharacters(tc.input)
+
+			if !slices.Equal(tc.want, got) {
+				t.Errorf("expected %v, got %v", tc.want, got)
+			}
+		})
+	}
+}
+
+func Test_computeFeedback(t *testing.T) {
+	tt := map[string]struct {
+		guess            string
+		solution         string
+		expectedFeedback Feedback
+	}{
+		"nominal": {
+			guess:            "HERTZ",
+			solution:         "HERTZ",
+			expectedFeedback: Feedback{correctPosition, correctPosition, correctPosition, correctPosition, correctPosition},
+		},
+		"double character": {
+			guess:            "HELLO",
+			solution:         "HELLO",
+			expectedFeedback: Feedback{correctPosition, correctPosition, correctPosition, correctPosition, correctPosition},
+		},
+		"double character with wrong answer": {
+			guess:            "HELLL",
+			solution:         "HELLO",
+			expectedFeedback: Feedback{correctPosition, correctPosition, correctPosition, correctPosition, absentCharacter},
+		},
+		"five identical, but only two are there": {
+			guess:            "LLLLL",
+			solution:         "HELLO",
+			expectedFeedback: Feedback{absentCharacter, absentCharacter, correctPosition, correctPosition, absentCharacter},
+		},
+		"two identical, but not in the right position (from left to right)": {
+			guess:            "HLLEO",
+			solution:         "HELLO",
+			expectedFeedback: Feedback{correctPosition, wrongPosition, correctPosition, wrongPosition, correctPosition},
+		},
+		"three identical, but not in the right position (from left to right)": {
+			guess:            "HLLLO",
+			solution:         "HELLO",
+			expectedFeedback: Feedback{correctPosition, absentCharacter, correctPosition, correctPosition, correctPosition},
+		},
+		"one correct, one incorrect, one absent (left of the correct)": {
+			guess:            "LLLWW",
+			solution:         "HELLO",
+			expectedFeedback: Feedback{wrongPosition, absentCharacter, correctPosition, absentCharacter, absentCharacter},
```

Comment:
Find a way to write the expected feedback in a shorter way, currently it's quite difficult to easily read it, e.g.

```
test("two identical, but not in the right position (from left to right)",
	"LLLWW",
	"HELLO",
	"?_+__"
)

// or

"one correct, one incorrect, one absent (left of the correct)": {
	guess:            "LLLWW",
	solution:         "HELLO",
	expectedFeedback: "?_+__",
}
```